# Nobile-Hernday Bioinformatics Workshop [Feb 19, 2019]
<br />

![alt text](https://raw.githubusercontent.com/sabahzero/Nobile-HerndayLabs_BioinformaticsWorkshop_Feb19-2019/master/Relevant-Articles_Figures-of-Interest-Highlighted/Nobile-Lab_UC-Merced.png) 
This is for an introduction tutorial on bioinformatics tailored to Nobile and Hernday lab members. <br />
Contributors: Sabah Ul-Hasan ([@sabahzero](https://github.com/sabahzero)), Akshay Paropkari ([@akshayparopkari](https://github.com/akshayparopkari)), Thaddeus Seher ([@tdseher](https://github.com/tdseher))

## Overview and aims of today
09:00 - 09:30 AM Introduction, ensuring everyone has Jupyter installed correctly on their computer </br>
09:30 - 10:30 AM Basics, creating bar plots with standard error, changing color schemes, etc </br>
10:30 - 11:00 AM Overview of the MERCED cluster, how to use it through command line, why and general Q&A </br>
</br>

### 09:00 - 09:30 AM Introduction, ensuring everyone has Jupyter installed correctly on their computer
1. Exposure to R, GitHub, and Jupyter as complementary tools for data analysis over using Excel </br>
&nbsp;&nbsp;&nbsp;a. What is R, and why use it as a scientist? Read introduction article [here](https://www.r-project.org/about.html) </br> 
&nbsp;&nbsp;&nbsp;b. The power of Git and Github as an open-source interface used with R: [Nature article](http://blogs.nature.com/naturejobs/2018/06/11/git-the-reproducibility-tool-scientists-love-to-hate/) </br> 
&nbsp;&nbsp;&nbsp;c. Using Jupyter notebook as a reproducible bioinformatics 'lab notebook': [Nature article](https://www.nature.com/articles/d41586-018-07196-1) </br>

### 09:30 - 10:30 AM Basics, creating bar plots with standard error, changing color schemes, etc
2. Using Jupyter notebook to run R code for RNA-Seq data analysis, and interacting with Gihtub as an interface </br> 
&nbsp;&nbsp;&nbsp;a. RNA-Seq analysis in R [resource](http://combine-australia.github.io/RNAseq-R/)</br> 
&nbsp;&nbsp;&nbsp;b. CHIP-Seq analysis in R [resource](https://link.springer.com/protocol/10.1007%2F978-1-4939-7380-4_17)</br> 
3. Using Jupyter notebook and R code to create publish-ready figures from an Excel datasheet for: </br>
&nbsp;&nbsp;&nbsp;a. Growth curves and scatter plots </br>
&nbsp;&nbsp;&nbsp;b. Bar charts, boxplots, and histograms with significance vs variance </br>
&nbsp;&nbsp;&nbsp;c. Heatmaps with significance vs variance </br>
&nbsp;&nbsp;&nbsp;d. Co-occurrence, network graphs, and venn diagrams with significance vs variance </br>

We will primarily be working with RNA-Seq Data </br>

### 10:30 - 11:00 AM Overview of the MERCED cluster, how to use it through command line, why and general Q&A </br>
</br>
